lighthouse/blueprints/plate_events.py

```diff
@@ -14,25 +20,37 @@
 @bp.route("/plate-events/create", methods=["GET"])
 def create_plate_event() -> Tuple[Dict[str, Any], int]:
     try:
-        barcode = request.args.get("barcode", "")
         event_type = request.args.get("event_type", "")
-        if len(barcode) == 0 or len(event_type) == 0:
-            return {
-                "errors": ["GET request needs 'barcode' and 'event_type' in url"]
-            }, HTTPStatus.BAD_REQUEST
-        else:
-            logger.info(
-                f"Attempting to create a '{event_type}' event for plate with barcode '{barcode}'"
+        logger.info(f"Attempting to publish an '{event_type}' plate event message")
+        if len(event_type) == 0:
+            logger.error(
+                "Failed publishing plate event message: missing required 'event_type' parameter"
+            )
+            return {"errors": ["'event_type' is a required parameter"]}, HTTPStatus.BAD_REQUEST
+
+        logger.info("Attempting to construct the plate event message")
+        errors, message = construct_event_message(event_type, request.args)
+        if len(errors) > 0:
+            logger.error(
+                "Failed publishing plate event message: error(s) constructing event message: "
+                f"{errors}"
             )
+            return {"errors": errors}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+        # By this stage we know the event type is valid as we have been able to construct a message
+        routing_key = get_routing_key(event_type)
+
+        logger.info("Attempting to publish the constructed plate event message")
+        broker = Broker()
+        broker.connect()
+        broker.publish(message, routing_key)
+        broker.close_connection()
```

Comment:
Maybe adding this close_connection in a way that even if publish raise an exception, it will be closed